src/viper/trans.ml

```diff
@@ -766,6 +830,22 @@ and rets t_opt =
 
 and id id = { it = id.it; at = id.at; note = NoInfo }
 
+(* Make sure that you call it before translating inner expressions.
+   This thing mutates [ctxt] environment and we want to update it first. *)
+and loop_label ctxt loop_label =
+  let label_name =
+    loop_label.it
+      |> String.map (fun ch ->
+          (* Life without Core is hard... *)
+          if Char.equal ch ' '
+          then '_'
+          else ch)
+      |> Format.asprintf "$%s"
```

Comment:
We use `$` to avoid conflicts between compiler-generated and user-written identifiers. Since the user chooses the name for the label, adding `$` in front of it does not fit this scheme.

Also I wonder if Motoko has different namespaces for labels and variables. That is, is it legal to have `label x` and `let x` in scope at the same time?

Reply:
In motoko labels and variables have different namespaces. However, in viper they share the same one. 
Right now I see 2 solutions:
1. Prefix it with `$lbl_`.
2. Handle it with `fresh_id` somehow. 

The latter sounds better but we should guarantee that `fresh_id` was called at least once for all `decl`s. 

Btw do we care about counters after variable names? Can't we just strip stamps map and use the global counter instead?

Reply:
Given a label `x`, I suggest we generate
1. `$lbl$break$x`
2. `$lbl$continue$x`

Or, if it's easier, `$lbl$x` and `$lbl$continue$x`

Reply:
> Or, if it's easier, $lbl$x and $lbl$continue$x

Supported this variant